mono/metadata/icall.c

```diff
@@ -5138,7 +5138,8 @@ ves_icall_System_Reflection_RuntimeAssembly_get_location (MonoReflectionAssembly
 {
 	MonoDomain *domain = MONO_HANDLE_DOMAIN (refassembly);
 	MonoAssembly *assembly = MONO_HANDLE_GETVAL (refassembly, assembly);
-	return mono_string_new_handle (domain, mono_image_get_filename (assembly->image), error);
+	const char *image_name = mono_image_get_filename (assembly->image);
+	return mono_string_new_handle (domain, strncmp (image_name, "data-", 5) != 0 ? image_name : "", error);
```

Comment:
Can we change `mono_image_get_filename` to return `image->filename` instead of `image->name`, instead?

Reply:
Oh, yep. Didn't realize that existed since it isn't set at all in the functions I was looking at.

Reply:
Since it's public API, I've opted to leave it for now and instead add some `m_image_get_` functions for internal use. Probably not worth the line damage to update existing callers to `mono_image_get_filename`, unless you think it's worth marking external-only.